aws/resource_aws_servicecatalog_provisioned_product.go

```diff
@@ -0,0 +1,370 @@
+	       input.Description = aws.String(v.(string))
+	   }
+	*/
+
+	log.Printf("[DEBUG] Provision Service Catalog Product: %#v", input)
+	resp, err := conn.ProvisionProduct(&input)
+	if err != nil {
+		return fmt.Errorf("Provisioning Service Catalog product failed: %s", err.Error())
+	}
+
+	d.SetId(*resp.RecordDetail.ProvisionedProductId)
+	if err := waitForServiceCatalogProvisionedProductStatus(conn, d); err != nil {
+		return err
+	}
+	return resourceAwsServiceCatalogProvisionedProductRead(d, meta)
+}
+
+func waitForServiceCatalogProvisionedProductStatus(conn *servicecatalog.ServiceCatalog, d *schema.ResourceData) error {
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{servicecatalog.ProvisionedProductStatusUnderChange},
+		Target:       []string{servicecatalog.ProvisionedProductStatusAvailable},
+		Refresh:      refreshProvisionedProductStatus(conn, d),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		PollInterval: 3 * time.Second,
+	}
+	_, err := stateConf.WaitForState()
+	return err
+}
+
+func refreshProvisionedProductStatus(conn *servicecatalog.ServiceCatalog, d *schema.ResourceData) resource.StateRefreshFunc {
+	return func() (result interface{}, state string, err error) {
+		resp, err := conn.DescribeProvisionedProduct(&servicecatalog.DescribeProvisionedProductInput{
+			Id: aws.String(d.Id()),
+		})
+		if err != nil {
+			// to help debug if there's a problem
+			d.Set("status", resp.ProvisionedProductDetail.Status)
+			d.Set("status_message", resp.ProvisionedProductDetail.StatusMessage)
+			return nil, "", err
+		}
+		return resp, aws.StringValue(resp.ProvisionedProductDetail.Status), nil
+	}
+}
+
+func waitForServiceCatalogProvisionedProductDeletion(conn *servicecatalog.ServiceCatalog, id string) error {
+	stateConf := resource.StateChangeConf{
+		Pending:      []string{servicecatalog.ProvisionedProductStatusUnderChange},
+		Target:       []string{""},
+		Timeout:      15 * time.Minute,
+		PollInterval: 3 * time.Second,
+		Refresh: func() (interface{}, string, error) {
+			resp, err := conn.DescribeProvisionedProduct(&servicecatalog.DescribeProvisionedProductInput{
+				Id: aws.String(id),
+			})
+			if err != nil {
+				if isAWSErr(err, servicecatalog.ErrCodeResourceNotFoundException, "") {
+					return 42, "", nil
+				}
+				return 42, "", err
+			}
+
+			return resp, aws.StringValue(resp.ProvisionedProductDetail.Status), nil
+		},
+	}
+	_, err := stateConf.WaitForState()
+	return err
+}
+
+func resourceAwsServiceCatalogProvisionedProductRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+
+	input := servicecatalog.DescribeProvisionedProductInput{
+		Id: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Reading Service Catalog Provisioned Product: %#v", input)
+	resp, err := conn.DescribeProvisionedProduct(&input)
+
+	if err != nil {
+		if scErr, ok := err.(awserr.Error); ok && scErr.Code() == "ResourceNotFoundException" {
```

Comment:
replace with `isAWSErr(err, servicecatalog.ErrCodeResourceNotFoundException, "")

Reply:
done (check-in to come)